playbooks/ansible-tox-py39/pre.yaml

```diff
@@ -1,8 +1,21 @@
 ---
 - hosts: all
+  vars:
+    python3_devel_pkg: python39-devel
   tasks:
-    - name: Ensure python3.9 is present
+
+    - name: Set python package name for Fedora 34
+      set_fact:
+        python3_devel_pkg: python3-devel
+      when: ansible_distribution == 'Fedora' and ansible_distribution_major_version == '34'
+
+    - name: Set python package name for Fedora 35
+      set_fact:
+        python3_devel_pkg: python3.9
+      when: ansible_distribution == 'Fedora' and ansible_distribution_major_version == '35'
```

Comment:
This is far more complicated than using a lookup and gets much harder to read, follow, and update as more distribution/version combinations are added. There is also more overhead and execution time involved since there needs to be one task per distro/version combination instead of one single task that just does a lookup.

```suggestion
    - name: Include distribution specific variables
      include_vars: "{{ lookup('first_found', params) }}"
      vars:
        params:
          files:
            - "{{ ansible_facts.distribution }}-{{ ansible_facts.distribution_version }}.yml"
            - "{{ ansible_facts.os_family }}-{{ ansible_facts.distribution_major_version }}.yml"
            - "{{ ansible_facts.distribution }}.yml"
            - "{{ ansible_facts.os_family }}.yml"
            - default.yml
          paths:
            - vars
```

```yaml
# vars/default.yml
python3_devel_pkg: python39-devel
```

```yaml
# vars/Fedora-34.yml
python3_devel_pkg: python3-devel
```

```yaml
# vars/Fedora-35.yml
python3_devel_pkg: python3.9
```